lib/pcr.c

```diff
@@ -211,6 +213,45 @@ bool pcr_print_pcr_struct(TPML_PCR_SELECTION *pcrSelect, tpm2_pcrs *pcrs) {
 }
 
 
+bool pcr_print_pcr_selection(TPML_PCR_SELECTION *pcrSelect) {
+    UINT32 i;
+    int j;
+    bool first;
+    const char* halgstr;
+    tpm2_tool_output ("SelectedPCRs:\n");
+
+    /* Iterate throught the pcr banks */
+    for (i = 0; i < pcrSelect->count; i++) {
+        /* Print hash alg of the current bank */
+        halgstr = tpm2_alg_util_algtostr(pcrSelect->pcrSelections[i].hash,
+                                         tpm2_alg_util_flags_hash);
+        if (halgstr != NULL) {
+            tpm2_tool_output("  - bank: %s\n", halgstr);
+        } else {
+            LOG_ERR("Unsupported hash algorithm 0x%08x", 
+                    pcrSelect->pcrSelections[i].hash);
+            return false;
+        }
+
+        /* Iterate through the PCRs of the bank */
+        first = true;
```

Comment:
No need for first, j == 0 is first.

Reply:
Don't use signed types unless you expect that things can go negative. Make j something other than int.
```C
sizeofSelect * 8
```
max math on that is 256 * 8 = 2040 which fits into 16 bits, so something like
```C
unsigned j;
```
would suffice.

Reply:
No, `j == 0` is not first... ;-)
First is first _set_ bit...